src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_pix2pix_zero.py

```diff
@@ -0,0 +1,843 @@
+        """
+        # 0. Define the spatial resolutions.
+        height = height or self.unet.config.sample_size * self.vae_scale_factor
+        width = width or self.unet.config.sample_size * self.vae_scale_factor
+
+        # 1. Check inputs. Raise error if not correct
+        self.check_inputs(
+            prompt,
+            self.conditions_input_image,
+            image,
+            source_embedding_path,
+            target_embedding_path,
+            callback_steps,
+            prompt_embeds,
+        )
+        if self.conditions_input_image and prompt_embeds:
+            logger.warning(
+                f"You have set `conditions_input_image` to {self.conditions_input_image} and"
+                " passed `prompt_embeds`. `prompt_embeds` will be ignored. "
+            )
+
+        # 2. Generate a caption for the input image if we are conditioning the
+        # pipeline based on some input image.
+        if self.conditions_input_image:
+            caption, preprocessed_image = generate_caption(image, self._captioner, self._captioner_processor)
+            height, width = preprocessed_image.shape[-2:]
+            prompt = caption
+            logger.info(f"Generated caption for the input image: {caption}.")
+
+        # 3. Define call parameters
+        if prompt is not None and isinstance(prompt, str):
+            batch_size = 1
+        elif prompt is not None and isinstance(prompt, list):
+            batch_size = len(prompt)
+        else:
+            batch_size = prompt_embeds.shape[0]
+        if cross_attention_kwargs is None:
+            cross_attention_kwargs = {}
+
+        device = self._execution_device
+        # here `guidance_scale` is defined analog to the guidance weight `w` of equation (2)
+        # of the Imagen paper: https://arxiv.org/pdf/2205.11487.pdf . `guidance_scale = 1`
+        # corresponds to doing no classifier free guidance.
+        do_classifier_free_guidance = guidance_scale > 1.0
+
+        # 3. Encode input prompt
+        prompt_embeds = self._encode_prompt(
+            prompt,
+            device,
+            num_images_per_prompt,
+            do_classifier_free_guidance,
+            negative_prompt,
+            prompt_embeds=prompt_embeds,
+            negative_prompt_embeds=negative_prompt_embeds,
+        )
+
+        # 4. Prepare timesteps
+        self.scheduler.set_timesteps(num_inference_steps, device=device)
+        timesteps = self.scheduler.timesteps
+
+        # 5. Generate the inverted noise from the input image or any other image
+        # generated from the input prompt.
+        if self.conditions_input_image:
+            # TODO (sayakpaul): Generate this using DDIM inversion.
+            # We need to get the inverted noise from the input image and this requires
+            # us to do a sort of `inverse_step()` in DDIM and then regularize the
+            # noise to enforce the statistical properties of Gaussian.
+            raise NotImplementedError
```

Comment:
I think we should raise an `error` early in the init. This will also avoid having to load the caption model and do inferences with it.